src/internal/storage/fileset/storage.go

```diff
@@ -69,7 +71,8 @@ func NewStorage(mds MetadataStore, tr track.Tracker, chunks *chunk.Storage, opts
 			FixedDelay:  DefaultCompactionFixedDelay,
 			LevelFactor: DefaultCompactionLevelFactor,
 		},
-		filesetSem: semaphore.NewWeighted(math.MaxInt64),
+		filesetSem:    semaphore.NewWeighted(math.MaxInt64),
```

Comment:
`MaxInt64`? Apparently this isn't used?

Reply:
I'm not following.

Reply:
If it has the maximum size then we probably won't ever reach the limit.

Reply:
It is the default, we overwrite it with the functional option.